File: app/src/main/java/com/rdev/tryp/blocks/favourite_drivers/FavouriteDriversFragment.java
package com.rdev.tryp.blocks.favourite_drivers;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import com.rdev.tryp.ContentActivity;
import com.rdev.tryp.R;
import com.rdev.tryp.firebaseDatabase.utils.DatabaseUtils;

import java.util.ArrayList;
import java.util.List;
import afu.org.checkerframework.checker.nullness.qual.NonNull;
import afu.org.checkerframework.checker.nullness.qual.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class FavouriteDriversFragment extends Fragment implements View.OnClickListener {

    private View root;
    private RecyclerView recyclerView;
    private Adapter adapter;
    private RelativeLayout confirmLayout;
    private int position;
    private ImageView backBtn;
    final List<FavoriteDriver> drivers = new ArrayList<>();

    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        root = inflater.inflate(R.layout.fragment_favourite_drivers, container, false);

        initView();
        return root;
    }

    class OnItemClickListener {
        void onItemClick(View view, int pos) {
            position = pos;
            switch (view.getId()) {
                case R.id.like_layout:
                    adapter.disLike(position);
                    showConfirm();
                    break;
            }
        }
    }

    private void initView() {
        backBtn = root.findViewById(R.id.back_btn);
        backBtn.setOnClickListener(this);

        recyclerView = root.findViewById(R.id.recycler_view);
        recyclerView.setLayoutManager(new GridLayoutManager(getContext(), 2));
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setNestedScrollingEnabled(false);

        ((ContentActivity) getActivity()).database.getFavoritedDrivers(driver -> {
            if(!DatabaseUtils.INSTANCE.arrayContainsDriverId(drivers, driver.getDriverId())){

                FavoriteDriver temp = new FavoriteDriver(driver.getFirstName(), driver.getLastName());
                temp.setId(driver.getDriverId());
                temp.setCategory(driver.getCategory());
                temp.setPhoto(driver.getImage());

                drivers.add(temp);
                adapter.notifyItemInserted(drivers.size()-1);

            }
        });

        OnItemClickListener listener = new OnItemClickListener();
        adapter = new Adapter(drivers, getContext(), listener);

        recyclerView.setAdapter(adapter);
        adapter.notifyDataSetChanged();

        View confirmButton = root.findViewById(R.id.confirm_button);
        confirmButton.setOnClickListener(this);

        confirmLayout = root.findViewById(R.id.confirm_layout);
        confirmLayout.setOnClickListener(this);
        hideConfirm();
    }

    private void showConfirm() {
        RelativeLayout.LayoutParams rel_btn = new RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
        confirmLayout.setLayoutParams(rel_btn);
    }

    private void hideConfirm() {
        RelativeLayout.LayoutParams rel_btn = new RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT, 0);
        confirmLayout.setLayoutParams(rel_btn);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.confirm_button:
                onPositiveClick();
                break;
            case R.id.confirm_layout:
                onNegativeClick();
                break;
            case R.id.back_btn:
                ((ContentActivity) getActivity()).goHome();
                break;
        }
    }

    private void onPositiveClick() {
        ((ContentActivity) getActivity()).database.addDriverToFavorite(drivers.get(position).getId(), false);
        adapter.delete(position);
        hideConfirm();
    }

    private void onNegativeClick() {
        adapter.like(position);
        hideConfirm();
    }

}

File: app/src/main/java/com/rdev/tryp/blocks/favourite_drivers/FavouriteDriversFragment.kt
package com.rdev.tryp.blocks.favourite_drivers

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import com.rdev.tryp.ContentActivity
import com.rdev.tryp.R
import com.rdev.tryp.blocks.helper.BaseFragment
import com.rdev.tryp.firebaseDatabase.utils.DatabaseUtils
import java.util.ArrayList
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.rdev.tryp.firebaseDatabase.AvailableDriversChanged
import com.rdev.tryp.firebaseDatabase.model.Driver

class FavouriteDriversFragment : BaseFragment(), View.OnClickListener {

    private var root: View? = null
    private var recyclerView: RecyclerView? = null
    private var adapter: Adapter? = null
    private var confirmLayout: RelativeLayout? = null
    private var position: Int = 0
    private var backBtn: ImageView? = null
    private val drivers: MutableList<FavoriteDriver> = ArrayList()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        root = inflater.inflate(R.layout.fragment_favourite_drivers, container, false)

        initView()

        return root
    }

    inner class OnItemClickListener {
        internal fun onItemClick(view: View, pos: Int) {
            position = pos
            when (view.id) {
                R.id.like_layout -> {
                    adapter?.disLike(position)
                    showConfirm()
                }
            }
        }
    }

    private fun initView() {
        backBtn = root?.findViewById(R.id.back_btn)
        backBtn?.setOnClickListener(this)

        recyclerView = root?.findViewById(R.id.recycler_view)
        recyclerView?.layoutManager = GridLayoutManager(context, 2)
        recyclerView?.itemAnimator = DefaultItemAnimator()
        recyclerView?.isNestedScrollingEnabled = false

        (activity as ContentActivity).database.getFavoritedDrivers(object : AvailableDriversChanged.GetData.Driver{
            override fun onCompleted(driver: Driver?) {
                if (!DatabaseUtils.arrayContainsDriverId(drivers, driver?.driverId)) {
                    val temp = FavoriteDriver(driver?.firstName, driver?.lastName)
                    temp.id = driver?.driverId
                    temp.category = driver?.category
                    temp.photo = driver?.image

                    drivers.add(temp)
                    adapter?.notifyItemInserted(drivers.size - 1)
                }
            }
        })

        val listener = OnItemClickListener()
        adapter = Adapter(drivers, context, listener)

        recyclerView?.adapter = adapter
        adapter?.notifyDataSetChanged()

        val confirmButton = root?.findViewById<View>(R.id.confirm_button)
        confirmButton?.setOnClickListener(this)

        confirmLayout = root?.findViewById(R.id.confirm_layout)
        confirmLayout?.setOnClickListener(this)
        hideConfirm()
    }

    private fun showConfirm() {
        val btnTemp = RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
        confirmLayout?.layoutParams = btnTemp
    }

    private fun hideConfirm() {
        val btnTemp = RelativeLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT, 0)
        confirmLayout?.layoutParams = btnTemp
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.confirm_button -> onPositiveClick()
            R.id.confirm_layout -> onNegativeClick()
            R.id.back_btn -> (activity as ContentActivity).goHome()
        }
    }

    private fun onPositiveClick() {
        (activity as ContentActivity).database.addDriverToFavorite(drivers[position].id, false)
        adapter?.delete(position)
        hideConfirm()
    }

    private fun onNegativeClick() {
        adapter?.like(position)
        hideConfirm()
    }

}